packages/fx-core/src/plugins/resource/bot/dotnet/plugin.ts

```diff
@@ -92,6 +120,44 @@ export class DotnetBotImpl extends TeamsBotImpl {
       if (botPassword) {
         appSettings = appSettings.replace(RegularExpr.botPassword, botPassword);
       }
+      if (includeAad) {
```

Comment:
If it is SSO related logic, can it be encapsulated as a separate library and ask Bot plugin to call it instead of putting the logic in Bot plugin? This will help to be more componentized for V3.

Reply:
@eriolchan That's a good point. Since we may have multiple components in V3 try to modify app settings file in parallel (just like bot and sso here), we expect core to support read and write for app settings file just like project settings. This part is still under discussion, and will update the logic here later if we have any updates.